### PR TITLE
fix pause event handling

### PR DESCRIPTION
**Pull request by atouminet**

### CODE REVIEW

File: player/mpv.go [hunk 1]
```diff
@@ -72,6 +74,8 @@ func (mpv *MPV) Initialize() (chan State, int) {
 
 	mpv.checkError(C.mpv_initialize(mpv.handle))
 
+    C.mpv_observe_property(mpv.handle, C.ulong(PROP_PAUSE), C.CString("pause"), C.MPV_FORMAT_FLAG)
```
Comment (emersion) [inline player/mpv.go]: Style: s/spaces/tabs/

File: player/mpv.go [hunk 2]
```diff
@@ -35,6 +35,8 @@ type MPV struct {
 
 var logLibMPV = flag.Bool("log-libmpv", false, "log output of libmpv")
 
+const PROP_PAUSE uint64 = 0
```
Comment (emersion) [inline player/mpv.go]: The proper type for this constant is `C.uint64_t`. With that change, we should be able to remove the casts.

File: player/mpv.go (hunk 1 above)
Comment (emersion) [inline player/mpv.go]: `C.CString` allocates memory which needs to be free'd via `C.free`.

File: player/mpv.go [hunk 3]
```diff
@@ -314,12 +318,17 @@ func (mpv *MPV) eventHandler(eventChan chan State) {
...
-		case C.MPV_EVENT_PAUSE:
-			eventChan <- STATE_PAUSED
-		case C.MPV_EVENT_UNPAUSE:
-			eventChan <- STATE_PLAYING
-		}
-	}
+        case C.MPV_EVENT_PROPERTY_CHANGE:
+            prop := (*C.mpv_event_property)(event.data)
+            if uint64(event.reply_userdata) == PROP_PAUSE {
+                if *(*int32)(prop.data) != 0 {
```
Comment (emersion) [inline player/mpv.go]: `MPV_FORMAT_FLAG` gives an `int`, which may not be `int32`. We should cast to `C.int` here instead.

File: player/mpv.go (hunk 2 above)
Comment (emersion) [inline player/mpv.go]: Nit: idiomatic Go uses camelCase for constants.

**Comment by atouminet** [comment]
Fixes #4

**Review by emersion — APPROVED** [review]
Looks Good To Me.

**Review by DucNg — APPROVED** [review]
Looks good thanks a lot @atouminet! Kindly appreciate you contribution to this project @emersion!

**Pull request merged**